src/Provider/Provider.Web/Controllers/ErrorController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using Esfa.Recruit.Provider.Web.Configuration;
+using Esfa.Recruit.Provider.Web.Configuration.Routing;
+using Esfa.Recruit.Provider.Web.Exceptions;
+using Esfa.Recruit.Provider.Web.ViewModels.Error;
+using Esfa.Recruit.Vacancies.Client.Domain.Exceptions;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Exceptions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Provider.Web.Controllers
+{
+    [AllowAnonymous]
+    public class ErrorController : Controller
+    {
+        private readonly ILogger<ErrorController> _logger;
+
+        public ErrorController(ILogger<ErrorController> logger)
+        {
+            _logger = logger;
+        }
+
+        [Route("error/{id?}")]
+        public IActionResult Error(int id)
+        {
+            ViewBag.IsErrorPage = true; // Used by layout to show/hide elements.
+
+            switch (id)
+            {
+                case 403:
+                    return AccessDenied();
+                case 404:
+                    return PageNotFound();
+                default:
+                    break;
+            }
+
+            return View(new ErrorViewModel { StatusCode = id, RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
+        }
+
+        [Route("error/handle")]
+        public IActionResult ErrorHandler()
+        {
+            // TODO: LWA - Need to store the ProviderIdentifier somewhere!!!
```

Comment:
Interesting, should a user be able to come into the provider site without a ukprn even for error pages?

Reply:
404